src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -0,0 +1,284 @@
+                diagnostics.Add(node, useSiteDiagnostics);
+            }
+
+            if (hasErrors)
+            {
+                ReportBinaryOperatorError(node, diagnostics, node.OperatorToken, left, right, resultKind);
+            }
+
+            PrepareBoolConversionAndTruthOperator(returnType, node, kind, diagnostics, out Conversion boolConversion, out UnaryOperatorSignature boolOperator);
+            return new TupleBinaryOperatorInfo.Single(convertedLeftType, convertedRightType, resultOperatorKind, resultMethod, boolConversion, boolOperator);
+        }
+
+        /// <summary>
+        /// If a element-wise binary operator returns a non-bool type, we will either:
+        /// - prepare a conversion to bool if one exists
+        /// - prepare a truth operator: op_false in the case of an equality (`a == b` will be lowered to `!((a == b).op_false)) or op_true in the case of inequality
+        /// </summary>
+        private void PrepareBoolConversionAndTruthOperator(TypeSymbol type, BinaryExpressionSyntax node, BinaryOperatorKind binaryOperator, DiagnosticBag diagnostics,
+            out Conversion boolConversion, out UnaryOperatorSignature boolOperator)
+        {
+            // Is the operand implicitly convertible to bool?
+
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            var boolean = GetSpecialType(SpecialType.System_Boolean, diagnostics, node);
+            var conversion = this.Conversions.ClassifyImplicitConversionFromType(type, boolean, ref useSiteDiagnostics);
+            diagnostics.Add(node, useSiteDiagnostics);
+
+            if (conversion.IsImplicit)
+            {
+                boolConversion = conversion;
+                boolOperator = default;
+                return;
+            }
+
+            // It was not. Does it implement operator true (or false)?
+
+            UnaryOperatorKind boolOpKind;
+            switch (binaryOperator)
+            {
+                case BinaryOperatorKind.Equal:
+                    boolOpKind = UnaryOperatorKind.False;
+                    break;
+                case BinaryOperatorKind.NotEqual:
+                    boolOpKind = UnaryOperatorKind.True;
+                    break;
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(binaryOperator);
+            }
+
+            LookupResultKind resultKind;
+            ImmutableArray<MethodSymbol> originalUserDefinedOperators;
+            BoundExpression comparisonResult = new BoundTupleOperandPlaceholder(node, type);
+            UnaryOperatorAnalysisResult best = this.UnaryOperatorOverloadResolution(boolOpKind, comparisonResult, node, diagnostics, out resultKind, out originalUserDefinedOperators);
+            if (best.HasValue)
+            {
+                boolConversion = best.Conversion;
```

Comment:
>boolConversion = best.Conversion; [](start = 16, length = 33)

This feels somewhat misleading, ```best.Conversion``` is not conversion to boolean, but conversion to operator's parameter type.